PowerShell/JumpCloud Module/Public/Systems/Get-JCSystemApp.ps1

```diff
@@ -43,298 +39,289 @@ function Get-JCSystemApp () {
+                                        [void]$resultsArrayList.Add($_)
+                                    }
+                                } elseif ($SystemOS) {
+                                    Get-JcSdkSystemInsightProgram -Filter @("name:eq:$SoftwareName") | ForEach-Object {
+                                        [void]$resultsArrayList.Add($_)
+                                    }
+                                }
                             } elseif ($SystemID) {
-                                $URL = "$JCUrlBasePath/api/v2/systeminsights/programs?filter=system_id:eq:$SystemID"
+                                if ($SoftwareVersion) {
+                                    Write-Error "Cannot search for software version on Windows without software name."
+                                } else {
+                                    Get-JcSdkSystemInsightProgram -Filter @("system_id:eq:$SystemID") | ForEach-Object {
+                                        [void]$resultsArrayList.Add($_)
+                                    }
+                                }
+                            } elseif ($SystemOS) {
+                                if ($SoftwareVersion) {
+                                    Write-Error "Cannot search for software version on Windows without software name."
+                                } else {
+                                    Get-JcSdkSystemInsightProgram | ForEach-Object {
+                                        [void]$resultsArrayList.Add($_)
+                                    }
+                                }
                             }
-                            Write-Debug $URL
                         }
                         'Darwin' {
-                            # If Software title, version, and system ID are passed then return specific app
-                            # If $softwareName does not have .app at the end then add it
-                            if ((!$SoftwareName) -and (!$SystemOs) -and (!$SoftwareVersion)) {
-                                # Add filter for system ID to $Search
-                                $URL = "$JCUrlBasePath/api/v2/systeminsights/apps"
-                            }
+
                             if ($SoftwareName) {
                                 # Check for .app at the end of the software name
+                                $macOsSoftwareName = $SoftwareName
                                 if (-not $SoftwareName.EndsWith('.app')) {
                                     Write-Debug "Adding .app to $SoftwareName"
                                     if ($SoftwareName.EndsWith('.App')) {
                                         Write-Debug "Replacing .App with .app"
-                                        $SoftwareName = $SoftwareName.Replace('.App', '.app')
+                                        $macOsSoftwareName = $macOsSoftwareName.Replace('.App', '.app')
```

Comment:
Per our conversation Friday, let's replace the .app replacement with our shorter version

`$softwareNameSubstring = $SoftwareName.Substring($SoftwareName.Length - 4)
If ($softwareNameSubstring -match '.app') {
    $SoftwareName = $SoftwareName.Replace($ending, $softwareNameSubstring.toLower())
} else {
    $softwareName = "$SoftwareName.app"
}`